java/base-client/src/main/java/no/bankaxept/epayment/client/base/BaseClient.java

```diff
@@ -28,7 +28,43 @@ public BaseClient(String baseurl, String apimKey, String username, String passwo
                 .findFirst()
                 .map(httpClientProvider -> httpClientProvider.create(baseurl))
                 .orElseThrow();
-        this.tokenPublisher = new AccessTokenProcessor("/bankaxept-epayment/access-token-api/v1/accesstoken", apimKey, username, password, clock, Executors.newScheduledThreadPool(1), httpClient);
+        this.tokenPublisher = new ScheduledAccessTokenPublisher("/bankaxept-epayment/access-token-api/v1/accesstoken", apimKey, username, password, clock, Executors.newScheduledThreadPool(1), httpClient);
+    }
+
+    private BaseClient(String baseurl, String token) {
+        httpClient = ServiceLoader.load(HttpClientProvider.class)
+                .findFirst()
+                .map(httpClientProvider -> httpClientProvider.create(baseurl))
+                .orElseThrow();
+        this.tokenPublisher = new StaticAccessTokenPublisher(token);
+    }
+
+    private BaseClient(String baseurl) {
+        httpClient = ServiceLoader.load(HttpClientProvider.class)
+                .findFirst()
+                .map(httpClientProvider -> httpClientProvider.create(baseurl))
+                .orElseThrow();
+        this.tokenPublisher = new EmptyAccessTokenPublisher();
+    }
+
+    private BaseClient(String baseurl, Supplier<String> tokenSupplier) {
+        httpClient = ServiceLoader.load(HttpClientProvider.class)
+                .findFirst()
+                .map(httpClientProvider -> httpClientProvider.create(baseurl))
+                .orElseThrow();
+        this.tokenPublisher = new SuppliedAccessTokenPublisher(tokenSupplier);
+    }
```

Comment:
This can be DRYed up a bit by using a common constructor on `(String baseUrl, AccessTokenPublisher publisher)`.